app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java

```diff
@@ -109,11 +112,19 @@ public void onActivityResult(final int requestCode, final int resultCode, final
             LOGGER.info("Adding new contact to database");
             //Name is the first part
             final String name = result.split(SEMI_COLON)[0];
-            //Key is the second part
-            final IKey key = QRGenerator.deconstructToKey(result.split(SEMI_COLON)[1]);
+            //IBAN is the second part
+            final String iban = result.split(SEMI_COLON)[1];
+            //Key is the third part
+            final IKey key = QRGenerator.deconstructToKey(result.split(SEMI_COLON)[2]);
 
             final IDatabase database = this.serviceLocator.getDatabase();
-            final Contact contact = new Contact(name, key);
+            Contact contact;
+            try {
+                contact = new Contact(name, key, new IBAN(iban));
+            } catch (IllegalArgumentException e) {
+                LOGGER.info("IBAN is not valid", e);
+                contact = new Contact(name, key);
```

Comment:
No, I don't think so. This error will probably not occur, because the IBAN of the other person needs to be valid. But if it does occur, it will log it and they can then add the IBAN manually through the ChangeContactActivity.